# Description: The Mesh11sd daemon sets the interface name 'mesh<index>' to the mesh interface.
# However, even after disabling the mesh mode, this interface name persists, leading to confusion
# when operating in other modes like AP/STA/Ad-Hoc, where the interface name continues to display as 'mesh<idx>'.
# This patch addresses the issue by removing the 'mesh<idx>' interface name from the configuration when the device
# is no longer in mesh mode. As this modification is not present in the upstream version, it is necessary to apply
# this local patch to resolve the inconsistency

diff --git a/src/mesh11sd b/src/mesh11sd
index 85041d3..5f50e97 100755
--- a/src/mesh11sd
+++ b/src/mesh11sd
@@ -387,45 +387,45 @@ elif [ "$1" = "daemon" ]; then
 
 		if [ $mode -eq 5 ]; then
 			# startup=4, statusmode=0, enabled=1
+			syslogmessage="mesh11sd v$version has started: mesh management mode $mode"
 			startup=0
 			statusmode=0
 			mode=1
-			syslogmessage="mesh11sd v$version has started: mesh management mode $mode"
 
 		elif [ $mode -eq 4 ]; then
 			# startup=4, statusmode=0, enabled=0
+			syslogmessage="mesh11sd v$version has started: mesh status mode $mode"
 			startup=0
 			statusmode=2
 			mode=0
-			syslogmessage="mesh11sd v$version has started: mesh status mode $mode"
 
 		elif [ $mode -eq 3 ]; then
 			# startup=0, statusmode=2, enabled=1
+			syslogmessage="mesh11sd v$version has started: mesh management mode $mode"
 			startup=0
 			statusmode=0
 			mode=1
-			syslogmessage="mesh11sd v$version has started: mesh management mode $mode"
 
 		elif [ $mode -eq 2 ]; then
 			# startup=0, statusmode=2, enabled=0
+			syslogmessage="mesh11sd v$version has started: mesh status mode $mode"
 			startup=0
 			statusmode=2
 			mode=0
-			syslogmessage="mesh11sd v$version has started: mesh status mode $mode"
 
 		elif [ $mode -eq 1 ]; then
 			# startup=0, statusmode=0, enabled=1
+			syslogmessage="mesh11sd v$version has started, mesh management mode $mode"
 			startup=0
 			statusmode=0
 			mode=1
-			syslogmessage="mesh11sd v$version has started, mesh management mode $mode"
 
 		elif [ $mode -eq 0 ]; then
 			# startup=0, statusmode=0, enabled=0
+			syslogmessage="mesh11sd v$version has started: mesh status mode $mode"
 			startup=0
 			statusmode=2
 			mode=0
-			syslogmessage="mesh11sd v$version has started: mesh status mode $mode"
 		fi
 
 		if [ $mode -ne $lastmode ]; then
